### ISSUE TITLE
no display of files and app crashes

### ISSUE THREAD

**Issue by Tealk** [nextcloud/android#7056]
You won’t see any files on the Nextcloud Android app. It doesn’t matter if you have a new connected account or an already connected account.
After a few seconds/minutes the error message below appears
3 Android devices have this error, 2 iOS devices run without problems

### Can you reproduce this problem on https://try.nextcloud.com?
No but strangely enough I have this behavior on 3 other instances running on v20

### Environment data
Nextcloud version: 20.0.0.9
Operating system and version: Debian 10
Apache or nginx version: nginx/1.18.0
PHP version: 7.4

#### App Error
```
************ CAUSE OF ERROR ************

java.util.ConcurrentModificationException
	at java.util.ArrayList.sort(ArrayList.java:1472)
	at java.util.Collections.sort(Collections.java:206)
	at com.owncloud.android.utils.FileSortOrderByName.sortCloudFiles(FileSortOrderByName.java:52)
	at com.owncloud.android.ui.adapter.OCFileListAdapter.swapDirectory(OCFileListAdapter.java:937)
	at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1275)
	at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1228)
	at com.owncloud.android.ui.fragment.OCFileListFragment.refreshDirectory(OCFileListFragment.java:1224)
	at com.owncloud.android.ui.activity.FileDisplayActivity.showFiles(FileDisplayActivity.java:2313)
	at com.owncloud.android.ui.activity.DrawerActivity.onNavigationItemClicked(DrawerActivity.java:332)
	at com.owncloud.android.ui.activity.DrawerActivity.lambda$null$0$DrawerActivity(DrawerActivity.java:291)
	at com.owncloud.android.ui.activity.-$$Lambda$DrawerActivity$GuTM4h_4QmmtMeB02KW202AQU4g.run(Unknown Source:4)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:7814)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1075)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30130190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: a50
Model: SM-A505FN
Id: QP1A.190711.020
Product: a50eea

************ FIRMWARE ************
SDK: 29
Release: 10
Incremental: A505FNXXU5BTH8
```

#### Nextcloud log (data/nextcloud.log)
```
{"reqId":"Lty38F6YaIRKczNpsl82","level":4,"time":"06.10.2020 13:13:19","remoteAddr":"80.187.106.146","user":"1e9fb3c8-a990-1036-844e-33b12343625b","app":"webdav","method":"PROPFIND","url":"/remote.php/webdav/","message":{"Exception":"TypeError","Message":"Argument 1 passed to OC\\Group\\Manager::getUserGroupIds() must implement interface OCP\\IUser, null given, called in /var/www/DOMAIN/lib/private/Share20/DefaultShareProvider.php on line 884","Code":0,"Trace":[{"file":"/var/www/DOMAIN/lib/private/Share20/DefaultShareProvider.php","line":884,"function":"getUserGroupIds","class":"OC\\Group\\Manager","type":"->"},{"file":"/var/www/DOMAIN/lib/private/Share20/Manager.php","line":1333,"function":"getSharedWith","class":"OC\\Share20\\DefaultShareProvider","type":"->"},{"file":"/var/www/DOMAIN/apps/dav/lib/Connector/Sabre/Node.php","line":323,"function":"getSharedWith","class":"OC\\Share20\\Manager","type":"->"},{"file":"/var/www/DOMAIN/apps/dav/lib/Connector/Sabre/FilesPlugin.php","line":372,"function":"getNoteFromShare","class":"OCA\\DAV\\Connector\\Sabre\\Node","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/PropFind.php","line":96,"function":"OCA\\DAV\\Connector\\Sabre\\{closure}","class":"OCA\\DAV\\Connector\\Sabre\\FilesPlugin","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/DOMAIN/apps/dav/lib/Connector/Sabre/FilesPlugin.php","line":374,"function":"handle","class":"Sabre\\DAV\\PropFind","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"handleGetProperties","class":"OCA\\DAV\\Connector\\Sabre\\FilesPlugin","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/Server.php","line":1063,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/Server.php","line":989,"function":"getPropertiesByNode","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/Server.php","line":1678,"function":"getPropertiesIteratorForPath","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/Server.php","line":1661,"function":"writeMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":363,"function":"generateMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPropFind","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/Server.php","line":474,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/Server.php","line":251,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/3rdparty/sabre/dav/lib/DAV/Server.php","line":319,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/apps/dav/appinfo/v1/webdav.php","line":84,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/DOMAIN/remote.php","line":167,"args":["/var/www/DOMAIN/apps/dav/appinfo/v1/webdav.php"],"function":"require_once"}],"File":"/var/www/DOMAIN/lib/private/Group/Manager.php","Line":325,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.1","version":"20.0.0.9"}
```

**Comment by Tealk** [comment]
I tried a little bit more and it looks like it is a problem when the user comes from LDAP. Local users do not have this problem.

**Comment by lolotiti06** [comment]
Hello,
I have exactly the same problem on a fresh new NextCloud 20.0 install.
I tested on 2 different phones. While I can see the files on the windows client or on a browser, the Android 3.13.1 shows no files. Pitcures are displayed correctly.
User is coming from LDAP.
In addition to the above, user picture is displayed correctly.
Thanks,
Laurent.

**Comment by sylikc** [comment]
ah that's what's happening... ok, I thought I had misconfigured something.

What logs do the developers need?

I have no problem on the web, or with webdav on Joplin.  Only the NC Android client can't see any files.  **It sees activity, and even photos, but no files.**  I tried resetting data in the app, relogging in, everything.  I tried rolling back to an old client and that didn't fix the problem either.

I'm also an LDAP user.   NC 20.0
